src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = item[key];
+
+    const keyString = String(keyValue);
+
+    if (!result[keyString]) {
+      result[keyString] = [];
+    }
```

Comment:
Consider using the 'in' operator to check for the existence of 'keyString' in 'result' for better readability and intent expression.